src/gluonts/model/seq2seq/_mq_dnn_estimator.py

```diff
@@ -14,112 +14,151 @@
-        mlp_hidden_dimension_seq: List[int] = list(),
-        quantiles: List[float] = list(),
+        prediction_length: int,
+        context_length: Optional[int] = None,
+        use_feat_dynamic_real: bool = False,
+        use_feat_static_cat: bool = False,
+        cardinality: List[int] = None,
+        embedding_dimension: List[int] = None,
+        add_time_feature: bool = False,
+        add_age_feature: bool = False,
+        seed: Optional[int] = None,
+        decoder_mlp_dim_seq: Optional[List[int]] = None,
+        channels_seq: Optional[List[int]] = None,
+        dilation_seq: Optional[List[int]] = None,
+        kernel_size_seq: Optional[List[int]] = None,
+        use_residual: bool = True,
+        quantiles: Optional[List[float]] = None,
         trainer: Trainer = Trainer(),
     ) -> None:
-        context_length = (
-            prediction_length if context_length is None else context_length
-        )
-        assert all(
-            [d > 0 for d in mlp_hidden_dimension_seq]
+
+        assert (
+            prediction_length > 0
+        ), f"Invalid prediction length: {prediction_length}."
+        assert decoder_mlp_dim_seq is None or all(
+            d > 0 for d in decoder_mlp_dim_seq
         ), "Elements of `mlp_hidden_dimension_seq` should be > 0"
+        assert channels_seq is None or all(
+            [d > 0 for d in channels_seq]
+        ), "Elements of `channels_seq` should be > 0"
+        assert dilation_seq is None or all(
+            [d > 0 for d in dilation_seq]
+        ), "Elements of `dilation_seq` should be > 0"
+        # TODO: add support for kernel size=1
+        assert kernel_size_seq is None or all(
+            [d > 1 for d in kernel_size_seq]
+        ), "Elements of `kernel_size_seq` should be > 0"
+        assert quantiles is None or all(
+            [0 <= d <= 1 for d in quantiles]
+        ), "Elements of `quantiles` should be >= 0 and <= 1"
+
+        self.decoder_mlp_dim_seq = (
+            decoder_mlp_dim_seq if decoder_mlp_dim_seq is not None else [20]
+        )
+        self.channels_seq = (
+            channels_seq if channels_seq is not None else [30, 30, 30]
+        )
+        self.dilation_seq = (
+            dilation_seq if dilation_seq is not None else [1, 3, 9]
+        )
+        self.kernel_size_seq = (
+            kernel_size_seq if kernel_size_seq is not None else [3, 3, 3]
+        )
+        self.quantiles = (
+            quantiles
+            if quantiles is not None
+            else [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
+        )
+
+        assert (
+            len(self.channels_seq)
+            == len(self.dilation_seq)
+            == len(self.kernel_size_seq)
+        ), (
+            f"mismatch CNN configurations: {len(self.channels_seq)} vs. "
+            f"{len(self.dilation_seq)} vs. {len(self.kernel_size_seq)}"
+        )
+
+        if seed:
+            np.random.seed(seed)
+            mx.random.seed(seed)
```

Comment:
Not sure we want that here. But I guess it's ok for now and we think about how to integrate this into gluon-ts generally.

Reply:
Bernie wanted some reproducibility, I guess thats why it was in here, but I agree that it does not really belong there long term.